docs/knowledgebase/advanced/account-info.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Account Information
+---
+
+## `Account` StorageMap
+
+In Substrate, account information is stored in a
+[**Storage Map**](https://substrate.dev/rustdocs/v3.0.0/frame_support/storage/types/struct.StorageMap.html#impl),
+extracted and shown below:
+
+[`frame/system/src/lib.rs`](https://github.com/paritytech/substrate/blob/bcd649ffca9efc93f8b4ac1506ec8117b71e1aac/frame/system/src/lib.rs#L530-L538):
+
+```rust
+/// The full account information for a particular account ID.
+#[pallet::storage]
+#[pallet::getter(fn account)]
+pub type Account<T: Config> = StorageMap<
+  _,
+  Blake2_128Concat,
+  T::AccountId,
+  AccountInfo<T::Index, T::AccountData>,
+  ValueQuery,
+>;
+```
+
+`Account` storage map takes in five type parameters, so it has no prefix, use `Blake2_128Concat` as
+the hash algorithm, use `T::AccountId`, the account public key as the map key, with
+`AccountInfo<T::Index, T::AccountData>` as the map value.
+
+## `AccountInfo` Structure
+
+Looking further down in the source code, we will see what `AccountInfo` is actually consisted of.
+
+[`frame/system/src/lib.rs`](https://github.com/paritytech/substrate/blob/bcd649ffca9efc93f8b4ac1506ec8117b71e1aac/frame/system/src/lib.rs#L787-L803):
+
+```rust
+#[derive(Clone, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode)]
+pub struct AccountInfo<Index, AccountData> {
+  /// The number of transactions this account has sent.
+  pub nonce: Index,
+  /// The number of other modules that currently depend on this account's existence. The account
+  /// cannot be reaped until this is zero.
+  pub consumers: RefCount,
+  /// The number of other modules that allow this account to exist. The account may not be reaped
+  /// until this and `sufficients` are both zero.
+  pub providers: RefCount,
+  /// The number of modules that allow this account to exist for their own purposes only. The
+  /// account may not be reaped until this and `providers` are both zero.
+  pub sufficients: RefCount,
+  /// The additional data that belongs to this account. Used to store the balance(s) in a lot of
+  /// chains.
+  pub data: AccountData,
+}
+```
+
+Every account is an `AccountInfo` consisting of an `nonce` indicating number of transactions the
+account has sent, three reference counters, and an `AccountData` structure which contains some
+balances ([further explained below](#accountdata-structure)).
+
+Let's look deeper into the three reference counters, `consumers`, `providers`, and `sufficients`.
+The reasons these counters exist because Substrate has a mechanism of slashing accounts (e.g. one
+reason could be an account balance is below the existential deposit). If any of these reference
+counters are greater than zero, it is telling Substrate to not slash the account. For:
```

Comment:
> it is telling Substrate to not slash the account.

it is telling Substrate that we should not allow the account to be destroyed, which usually means the user trying to transfer their existential deposit away